qp_klp/tests/test_step.py

```diff
@@ -991,6 +991,24 @@ def test_precheck(self):
         with self.assertRaisesRegex(PipelineError, msg):
             step.precheck(fake_client)
 
+    def test_project_metadata_check(self):
+        fake_client = FakeClient()
+
+        # self.pipeline represents a metagenomic pathway.
+        step = Step(self.pipeline, self.qiita_id, None)
+
+        # _project_metadata_check() should return w/out raising an Error if
+        # step and fake_client is used.
+        step._project_metadata_check(fake_client)
+
+        fake_client.info_in_11661['categories'].append('well_id_384')
+        fake_client.info_in_13059['categories'].append('well_id_384')
+
+        msg = ("'well_id_384' exists in Qiita study 13059's sample metadata"
+               "\n'well_id_384' exists in Qiita study 11661's sample metadata")
```

Comment:
My concern with this is that we are not actually testing the connection to qiita as you are using the fake_client, would it be possible to create a real one? Note that all plugins use this. See: https://github.com/qiita-spots/qp-woltka/blob/main/qp_woltka/tests/test_woltka.py#L26

Reply:
After discussion offline, @charles-cowart confirmed that he's pointing to the right endpoint cause he tested outside of the tests + we agreed that reintroducing actual testing against the local/dev version will be in a different PR.